Crash Log: Recreated Deleted VPN Gateway

It appears someone deleted the wrong VPN Gateway in the console when we tried to rerun Terraform, it crashed with the log attached. The fix was:

1. Taint the VPN Gateway in TF
2. Remove all references to the vpn gateway in the tfstate 
3. also any static routes as well.

Steps to reproduce:
1. Create a VPN Gateway via Terraform with some static routes
2. Delete the VPN Gateway in AWS Console
3. Run Terraform again

I am not sure how much this is Terraform's fault, but ideally this person should have deleted the VPN Gateway via Terraform to handle this gracefully imo.

I have the full crash.log if necessary, I just don't feel particularly comfortable sharing the whole log publicly. Below is what I found interesting (after the memory error, we noticed everything was tossing a EOF error):

`117390 2016/03/15 17:20:19 [TRACE] [walkRefresh] Entering eval tree: aws_route53_record.stub
117391 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: 2016/03/15 17:20:19 [DEBUG] launch configuration describe configuration: {
117392 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:   LaunchConfigurationNames: ["terraform-hjyk4kfnxjfkrbco75jspshkty"]
117393 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: }
117394 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: panic: runtime error: invalid memory address or nil pointer dereference
117395 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: [signal 0xb code=0x1 addr=0x0 pc=0x20ef0a]
117396 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:
117397 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: goroutine 2028 [running]:
117398 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: panic(0xcea400, 0xc82000a0e0)
117399 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
117400 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsVpnConnectionRead(0xc820293380, 0xa0c360, 0xc8200b4300, 0x0, 0x0)
117401 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /private/tmp/terraform20160227-37732-w3ykp1/terraform-0.6.12/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_vpn_connection.go:308 +0x85a
117402 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc8204d0f40, 0xc820341410, 0xa0c360, 0xc8200b4300, 0xc8202680a8, 0x0, 0x0)
117403 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /private/tmp/terraform20160227-37732-w3ykp1/terraform-0.6.12/src/github.com/hashicorp/terraform/helper/schema/resource.go:209 +0x43a
117404 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc82049efc0, 0xc820499a40, 0xc820341410, 0x790301, 0x0, 0x0)
117405 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /private/tmp/terraform20160227-37732-w3ykp1/terraform-0.6.12/src/github.com/hashicorp/terraform/helper/schema/provider.go:187 +0x1da
117406 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Refresh(0xc8204d88c0, 0xc8205d1fb0, 0xc8204a0d30, 0x0, 0x0)
117407 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /private/tmp/terraform20160227-37732-w3ykp1/terraform-0.6.12/src/github.com/hashicorp/terraform/rpc/resource_provider.go:345 +0x6a
117408 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: reflect.Value.call(0xc38500, 0xed7418, 0x13, 0xf79f60, 0x4, 0xc820755ed8, 0x3, 0x3, 0x0, 0x0, ...)
117409 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /usr/local/Cellar/go/1.6/libexec/src/reflect/value.go:435 +0x120d
117410 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: reflect.Value.Call(0xc38500, 0xed7418, 0x13, 0xc820755ed8, 0x3, 0x3, 0x0, 0x0, 0x0)
117411 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /usr/local/Cellar/go/1.6/libexec/src/reflect/value.go:303 +0xb1
117412 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: net/rpc.(*service).call(0xc8204d1040, 0xc8204d1000, 0xc8202b4e10, 0xc820194380, 0xc8204bc1e0, 0xa0d4a0, 0xc8205d1fb0, 0x16, 0xa0d500, 0xc8204a0d30, ...)
117413 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /usr/local/Cellar/go/1.6/libexec/src/net/rpc/server.go:383 +0x1c2
117414 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws: created by net/rpc.(*Server).ServeCodec
117415 2016/03/15 17:20:19 [DEBUG] terraform-provider-aws:     /usr/local/Cellar/go/1.6/libexec/src/net/rpc/server.go:477 +0x49d
117416 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalRefresh, err: aws_eip.dns: unexpected EOF
117417 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalSequence, err: aws_eip.dns: unexpected EOF
117418 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalOpFilter, err: aws_eip.dns: unexpected EOF
117419 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalRefresh, err: aws_vpn_connection.eng_lab_vpn: unexpected EOF
117420 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalSequence, err: aws_vpn_connection.eng_lab_vpn: unexpected EOF
117421 2016/03/15 17:20:19 [ERROR] root: eval: *terraform.EvalRefresh, err: aws_route53_record.nat_public: unexpected EOF`

--- a/builtin/providers/aws/resource_aws_vpn_connection.go
+++ b/builtin/providers/aws/resource_aws_vpn_connection.go
@@ -305,11 +305,13 @@ func resourceAwsVpnConnectionRead(d *schema.ResourceData, meta interface{}) erro
 	// Set read only attributes.
 	d.Set("customer_gateway_configuration", vpnConnection.CustomerGatewayConfiguration)
 
-	tunnelInfo := xmlConfigToTunnelInfo(*vpnConnection.CustomerGatewayConfiguration)
-	d.Set("tunnel1_address", tunnelInfo.Tunnel1Address)
-	d.Set("tunnel1_preshared_key", tunnelInfo.Tunnel1PreSharedKey)
-	d.Set("tunnel2_address", tunnelInfo.Tunnel2Address)
-	d.Set("tunnel2_preshared_key", tunnelInfo.Tunnel2PreSharedKey)
+	if vpnConnection.CustomerGatewayConfiguration != nil {
+		tunnelInfo := xmlConfigToTunnelInfo(*vpnConnection.CustomerGatewayConfiguration)
+		d.Set("tunnel1_address", tunnelInfo.Tunnel1Address)
+		d.Set("tunnel1_preshared_key", tunnelInfo.Tunnel1PreSharedKey)
+		d.Set("tunnel2_address", tunnelInfo.Tunnel2Address)
+		d.Set("tunnel2_preshared_key", tunnelInfo.Tunnel2PreSharedKey)
+	}
 
 	if err := d.Set("vgw_telemetry", telemetryToMapList(vpnConnection.VgwTelemetry)); err != nil {
 		return err

